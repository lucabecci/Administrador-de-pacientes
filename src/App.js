import React, { useState, useEffect } from 'react';
import Formulario from './components/Formulario';
import Cita from './components/Cita';

function App() {

  //Citas en localStorage
  let citasIniciales = JSON.parse(localStorage.getItem('citas'));
  if(!citasIniciales){
    citasIniciales = [];
  }
  //Arreglo de citas
  const [citas, guardarCitas] = useState(citasIniciales);

  //UseEffect para realizar ciertas operaciones cuando el state cambia
  useEffect( () =>{
    if(citasIniciales){
      localStorage.setItem('citas', JSON.stringify(citas))
    }else{
      localStorage.setItem('citas', JSON.stringify([]))
    }
  }, [citas, citasIniciales])
  //Funcion que tome las cita actuales y agrege una nueva
  const crearCita = cita =>{
    guardarCitas([...citas, cita]);
  }

  //Funcion que elimina una cita por su  ID
  const eliminarCita = id =>{
    const nuevasCitas = citas.filter(cita => cita.id !== id);
    guardarCitas(nuevasCitas)
  }

  //Mensaje condicional de administra tus citas
  const titulo = citas.length === 0 ? 'No hay citas': 'Administra tus citas'; 
  return (
    <>
    <h1>Administrador De Pacientes</h1>

    <div className ='container'>
      <div className ='row'>
        <div className = 'one-half column'>
            <Formulario 
              crearCita = {crearCita}
            />
        </div>
        <div className = 'one-half column'>
            <h2>{titulo}</h2>
            { citas.map( cita => (
              <Cita
              key = {cita.id}
              cita={cita}
              eliminarCita={eliminarCita}
              />
            ))}
        </div>
      </div>
    </div>
    </>
  );
}

export default App;
