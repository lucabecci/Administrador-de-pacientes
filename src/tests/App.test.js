import React from 'react'
import App from '../App'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

describe('Tests en App', () => {
    let wrapper = shallow( <App /> )
    test('Snapshot en App.js', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})
