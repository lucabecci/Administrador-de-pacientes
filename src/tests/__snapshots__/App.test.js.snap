// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en App Snapshot en App.js 1`] = `
<Fragment>
  <h1>
    Administrador De Pacientes
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="one-half column"
      >
        <Formulario
          crearCita={[Function]}
        />
      </div>
      <div
        className="one-half column"
      >
        <h2>
          No hay citas
        </h2>
      </div>
    </div>
  </div>
</Fragment>
`;
